sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_registry_client.py

```diff
@@ -424,11 +424,11 @@ def get_manifest_properties(self, repository: str, tag_or_digest: str, **kwargs)
         manifest_properties = self._client.container_registry.get_manifest_properties(
             repository, tag_or_digest, **kwargs
         )
-        return ArtifactManifestProperties._from_generated(  # pylint: disable=protected-access
-            manifest_properties.manifest,  # type: ignore
-            repository_name=repository,
-            registry=self._endpoint,
-        )
+        if manifest_properties.manifest is not None:
+            return ArtifactManifestProperties._from_generated(  # pylint: disable=protected-access
+                manifest_properties.manifest, repository_name=repository, registry=self._endpoint
+            )
+        return ArtifactManifestProperties()
```

Comment:
This will not work, as we have already typed `ArtifactManifestProperties` with some attributes being required.
Under what conditions would `manifest_properties.manifest` be `None`?

Reply:
mypy think manifest_properties.manifest could be None even though it's specified to be required in docstring, the None probably is from the default value in it's constructor. It looks we have to ignore the error in this scenario.

Reply:
Yeah that sounds like a bug in either the swagger spec or code gen. If `manifest` is required to be returned, we shouldn't have to check for `None` here.